From 5f5c37cf59afd73321de3bf0ebce96b659ecb3c2 Mon Sep 17 00:00:00 2001
From: Jyri Sarha <jsarha@ti.com>
Date: Mon, 16 Mar 2015 18:08:13 +0200
Subject: [PATCH 107/800] ASoC: davinci-evm: Set dai sysclk at init time on
 beagle-bone-black

Set dai sysclk at init time for beagle-bone-black HDMI audio. For the
davinci-mcasp implicit BCLK settings to work correctly the sysclk
should be set before stream startup.

Signed-off-by: Jyri Sarha <jsarha@ti.com>
[zou: Original patch taken from PROCESSOR-SDK-LINUX-AM335X 02_00_01_07]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 sound/soc/davinci/davinci-evm.c |   22 ++++++++--------------
 1 files changed, 8 insertions(+), 14 deletions(-)

diff --git a/sound/soc/davinci/davinci-evm.c b/sound/soc/davinci/davinci-evm.c
index c4a04ba..bf74ca0 100644
--- a/sound/soc/davinci/davinci-evm.c
+++ b/sound/soc/davinci/davinci-evm.c
@@ -86,25 +86,13 @@ static int evm_tda998x_startup(struct snd_pcm_substream *substream)
 	struct snd_pcm_runtime *runtime = substream->runtime;
 	struct snd_mask *fmt = constrs_mask(&runtime->hw_constraints,
 					    SNDRV_PCM_HW_PARAM_FORMAT);
+
 	snd_mask_none(fmt);
 	snd_mask_set(fmt, TDA998X_SAMPLE_FORMAT);
 
 	return evm_startup(substream);
 }
 
-static int evm_tda998x_hw_params(struct snd_pcm_substream *substream,
-				 struct snd_pcm_hw_params *params)
-{
-	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
-	struct snd_soc_card *soc_card = rtd->card;
-	struct snd_soc_card_drvdata_davinci *drvdata =
-		snd_soc_card_get_drvdata(soc_card);
-
-	return snd_soc_dai_set_sysclk(cpu_dai, 0, drvdata->sysclk,
-				      SND_SOC_CLOCK_IN);
-}
-
 static struct snd_soc_ops evm_ops = {
 	.startup = evm_startup,
 	.shutdown = evm_shutdown,
@@ -115,7 +103,6 @@ static struct snd_soc_ops evm_ops = {
 static struct snd_soc_ops evm_tda998x_ops = {
 	.startup = evm_tda998x_startup,
 	.shutdown = evm_shutdown,
-	.hw_params = evm_tda998x_hw_params,
 };
 
 /* davinci-evm machine dapm widgets */
@@ -186,12 +173,19 @@ static int evm_tda998x_init(struct snd_soc_pcm_runtime *rtd)
 	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
 	struct snd_soc_dapm_context *dapm = &rtd->codec->dapm;
 	struct snd_soc_card *soc_card = rtd->card;
+	struct snd_soc_card_drvdata_davinci *drvdata =
+		snd_soc_card_get_drvdata(soc_card);
 	int ret;
 
 	ret = snd_soc_dai_set_clkdiv(cpu_dai, 0, 1);
 	if (ret < 0)
 		return ret;
 
+	ret = snd_soc_dai_set_sysclk(cpu_dai, 0, drvdata->sysclk,
+				     SND_SOC_CLOCK_IN);
+	if (ret < 0)
+		return ret;
+
 	snd_soc_dapm_new_controls(dapm, tda998x_dapm_widgets,
 				  ARRAY_SIZE(tda998x_dapm_widgets));
 
-- 
1.7.5.4

